### PR TITLE
Expose partition ID for checkpointing

### PR DESCRIPTION
**Pull request by pingihu**

### CODE REVIEW

File: distributing_producer.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 // partition, returning the post-commit offset and any error encountered. The
 // offset of each message is also updated accordingly.
 type DistributingProducer interface {
-	Distribute(topic string, messages ...*proto.Message) (offset int64, err error)
+	Distribute(topic string, messages ...*proto.Message) (offset int64, partition int32, err error)
```
Comment (zorkian) [inline distributing_producer.go]: Flip the order -- the partition is the "more important" bit so it should go first.

File: distributing_producer.go [hunk 2]
```diff
@@ -83,7 +83,8 @@ func NewErrorAverseRRProducer(conf *errorAverseRRProducerConf) DistributingProdu
 		}}
 }
 
-func (d *errorAverseRRProducer) Distribute(topic string, messages ...*proto.Message) (offset int64, err error) {
+func (d *errorAverseRRProducer) Distribute(topic string, messages ...*proto.Message) (
+	offset int64, partition int32, err error) {
 	if count, err := d.partitionCountSource.PartitionCount(topic); err == nil {
```
Comment (zorkian) [inline distributing_producer.go:89]: nit: Blank line between multi-line declarations and code.

**Comment by zorkian** [comment]
Commented.

File: distributing_producer.go (hunk 2 above)
Comment (pingihu) [inline distributing_producer.go:89]: Oh derp, fixed.

**Comment by zorkian** [comment]
LGTM can you squash so I can merge?

**Pull request merged**